Add rules for Release Notes

rh-hemartin opened the pull request:
<!--  Thanks for sending a pull request!

Please read our contributor guidelines:
https://kubearchive.github.io/kubearchive/main/contributors/guide.html
-->

## Which Issue(s) this Pull Request Resolves
<!-- Please make sure to create an issue you can link to and mention it here
to automatically close it. If this PR covers part of the work, instead of
"Resolves #", use "Related to #"

Usage: `Resolves #<issue number>`, or `Resolves (paste link of issue)`.
-->

Resolves #944 

## Release Note
<!--
If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other KubeArchive contributors. If this
change has no user-visible impact, leave the code block as is.
-->

```release-note
NONE
```

## Notes for Reviewers
<!--
Leave notes for the reviewers if you want to point their attention to some
specific place.
-->

<!--
Please label this pull request according to what type of issue you are addressing.
For reference on required PR/issue labels, read here:
https://kubearchive.github.io/kubearchive/main/contributors/release.html#_pull_request_labels

Add one of the following labels:
kind/bug
kind/documentation
kind/feature

Optionally add one or more of the following kinds if applicable:
kind/breaking
-->

maruiz93 commented on .github/pull_request_template.md:
```diff
@@ -20,8 +20,11 @@ If this change has user-visible impact, write a release note in the block
 
-Write as if you are speaking to users, not other KubeArchive contributors. If this
-change has no user-visible impact, leave the code block as is.
+Write as if you are speaking to users, not other KubeArchive contributors. If this change has no user-visible impact, leave the code block as is.
```
This line is too long, I suggest splitting it in two lines:
```suggestion
Write as if you are speaking to users, not other KubeArchive contributors. 
If this change has no user-visible impact, leave the code block as is.
```

maruiz93 commented on docs/modules/ROOT/pages/contributors/release.adoc:
```diff
@@ -81,6 +81,29 @@ NONE
+
+=== Examples of Release Notes
+
+* Adds pprof endpoints enabled by default.
```
How about allowing links on release notes?

E.g:

```suggestion
* Adds [pprof](https://pkg.go.dev/net/http/pprof) endpoints enabled by default.
```

rh-hemartin replied:
Changed.

maruiz93 commented on docs/modules/ROOT/pages/contributors/release.adoc:
```diff
@@ -81,6 +81,29 @@ NONE
+* Adds pprof endpoints enabled by default.
+* Changes the default log returned from the first container to the one specified in
+the annotation `kubectl.kubernetes.io/default-container`.
+* Allows users to filter resources by label on the majority of API endpoints.
```
This could be more concise?
```suggestion
* Adds optional [labelSelector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) query parameter in most of the API endpoints.
```

rh-hemartin replied:
Changed.

maruiz93 commented on docs/modules/ROOT/pages/contributors/release.adoc:
```diff
@@ -81,6 +81,29 @@ NONE
+* Adds retries when resource updates are not processed correctly.
+* Fixes a bug where resource updates were always rejected, causing performance problems.
+* Improves API speed when requesting resources cluster-wide.
+
```
I think it would be useful to also mention bad examples of release notes:

* Vague
* Using technical language
* With fillers that makes it longer than needed

rh-hemartin replied:
Working on it.

rh-hemartin replied:
I added a couple of examples.

skoved commented on .github/pull_request_template.md:
```diff
@@ -20,8 +20,11 @@ If this change has user-visible impact, write a release note in the block
+Write as if you are speaking to users, not other KubeArchive contributors. If this change has no user-visible impact, leave the code block as is.
+
+See
+http://localhost:8080/release-notes-docs/contributors/release.html#_writing_release_notes
```
Should this link to the documentation we host on github?

```suggestion
https://kubearchive.github.io/kubearchive/main/contributors/release.html#_writing_release_notes
```

rh-hemartin replied:
Yes, good catch.

ggallen commented:
What exactly are the expectations around release notes?  This seems to deal with the content of a release note, if someone chose to write one.

But will a release note be required for every PR?  If not, what determines whether or not there should be a release note?  And who's going to monitor it?

rh-hemartin commented:
We talked about this on our meeting, but for the sake of completion I'll answer.

> What exactly are the expectations around release notes? This seems to deal with the content of a release note, if someone chose to write one.
>
> But will a release note be required for every PR? If not, what determines whether or not there should be a release note? And who's going to monitor it?

A release note is not required for every PR, there is a list of scenarios where they are required (verbatim from the documentation):

```
Some examples of changes that would require release notes are:

* Optimized the code increasing the overall speed by a signifcant amount.
* Fixed bug where some configuration option was ignored.
* Introduced a new configuration option.
* Rewrote a whole documentation page that is relevant to users.
* Added a documentation page covering a topic relevant to users.

Some examples of changes that would **not** require release notes are:

* Fixed typos on a documentation page.
* Improvements on contributor relevant topics such as test coverage, maintainability,
naming conventions...
* Reduced time to perform integration testing.
* Added missing tests.
```

KubeArchive's maintainers need to monitor if a PR is worth a release note, so when we review a PR we need to think if it needs a release note or if the release note is not properly written.

rh-hemartin commented:
@skoved could you take another look?

skoved commented:
my bad. taking a look now
